Marlin/src/gcode/lcd/M145.cpp

```diff
@@ -46,8 +46,10 @@ void GcodeSuite::M145() {
   else {
     preheat_t &mat = ui.material_preset[material];
     #if HAS_HOTEND
-      if (parser.seenval('H'))
-        mat.hotend_temp = constrain(parser.value_int(), EXTRUDE_MINTEMP, thermalManager.hotend_max_target(0));
+      if (parser.seenval('H')) {
+        mat.hotend_temp = parser.value_int();
+        LIMIT(mat.hotend_temp, EXTRUDE_MINTEMP, thermalManager.hotend_max_target(0));
```

Comment:
Is `constrain` not usable with the new type? How can we alert future developers who try to use it that they will get back garbage?

Reply:
It's probably usable. That's an error here, from the very beginning of the code. It should read:
`mat.hotend_temp = constrain(parser.value_celsius(), EXTRUDE_MINTEMP, thermalManager.hotend_max_target(0));`